### PR TITLE
Add savefig to circle graphs for #4714

### PR DESCRIPTION
**Pull request by jhouck**
For plot_mne_inverse_label_connectivity.py this only adds the explanation and commented commands.  For plot_mixed_source_space_connectivity.py I tweaked things a bit so that plt is used even when the savefig line is commented out.

### CODE REVIEW

**Comment by larsoner** [comment]
Can you update/add a `Notes` section for `plot_connectivity_circle`, too?

**Comment by jhouck** [comment]
Travis is complaining about two things: 1) an error loading the megsim dataset in datasets/test/test_datasets.py and 2) My example command inside a note in mne/viz/circle.py:146 .  I think the second is from the indent, not sure what the issue is with megsim.

File: mne/viz/circle.py [hunk 1]
```diff
@@ -134,8 +134,19 @@ def plot_connectivity_circle(con, node_names, indices=None, n_lines=None,
...
+    Notes: 
+    
+    This code is based on the circle graph example by Nicolas P. Rougier
     http://www.labri.fr/perso/nrougier/coding/.
+    
+    By default, ``matplotlib`` does not take ``facecolor`` into account, even
+    if set when a figure is generated. This can be addressed via 
+    :func:`matplotlib.pyplot.savefig`, e.g.::
+    
+    >>> fig.savefig(fname_fig, facecolor='black')
```
Comment (agramfort) [inline mne/viz/circle.py]: this might break the doc as it could be seen as a docstring to test

Comment (jhouck) [reply mne/viz/circle.py]: I see, so add # doctest:+SKIP to the end of the line as in construct_iir_filter in filter.py?

**Comment by agramfort** [comment]
yes

**Comment by jhouck** [comment]
And now I have a suggestion for the MNE 'getting started' pages:  recommend that those using spyder enable style analysis and the 'automatically remove trailing spaces when saving files' option.  Unless that's part of the hazing... ;)

**Comment by larsoner** [comment]
> And now I have a suggestion for the MNE 'getting started' pages

PR welcome :)

If you want you can even add it to this PR, or just start another. Presumably here?

https://github.com/mne-tools/mne-python/blob/master/doc/contributing.rst

**Comment by larsoner** [comment]
Merging after PEP8 fixes, thanks @jhouck

**Pull request merged**